### ISSUE TITLE
Do not set HierarchicalLedgerManagerFactory explicitly

### ISSUE THREAD

**Issue by eolivelli** [apache/pulsar#3457]
**Is your feature request related to a problem? Please describe.**
I would like to deploy a Pulsar cluster on an existing "old" BookKeeper cluster which is using the legacy FlatLedgerManagerFactory

**Describe the solution you'd like**
We can drop any direct reference to HierarchicalLedgerManagerFactory as it is the default in BookKeeper

**Describe alternatives you've considered**
Make the LedgerManagerFactory configurable

**Additional context**
In the future while migrating to metadataServiceUri we should use zk+null:/server;server2/ledgers syntax in order to be able to auto detect layout

**Comment by eolivelli** [comment]
@merlimat  @sijie  @ivankelly 

I can send a PR, it should be straigthforward

**Comment by sijie** [comment]
Enrico, +1 from me. Feel free to submit a pull request.

**Comment by rdhabalia** [comment]
what is the reason for not using `HierarchicalLedgerManagerFactory`?

**Comment by rdhabalia** [comment]
>  And people also are also using the new LongHierarchical for ledger id beyond 2^32.

Yes, I agree. So, can we keep `LongHierarchical` default option?

**Comment by sijie** [comment]
@rdhabalia 

> Yes, I agree. So, can we keep LongHierarchical default option?

we can use LongHierarchical as default option. however that is not very useful for zookeeper based metadata manager, because zookeeper's sequence id will not go beyond 2^32 :)

**Comment by eolivelli** [comment]
My idea for the patch that I am going to send is to drop every explicit setting of the LMF, this way we will pick the default one on new clusters and let the system work on any layout supported by the version of BK bundled with Pulsar.

The alternative is to make it configurable

**Comment by merlimat** [comment]
> we can use LongHierarchical as default option. however that is not very useful for zookeeper based metadata manager, because zookeeper's sequence id will not go beyond 2^32 :)

@sijie  I'm actually a bit confused on the difference between `LongHierarchicalLedgerManagerFactory` and `HierarchicalLedgerManager`. As I understand `HierarchicalLedgerManager` is already handling the 64bit ledger id generation by having the 2 parts: 

https://github.com/apache/bookkeeper/blob/e921193b464bc86b24fa63bb71ba85ebb2d523c6/bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/HierarchicalLedgerManagerFactory.java#L34-L41

@eolivelli is the BK client picking up the the ledger manager implementation from the ZK metadata unless specified?

One other part to be careful is the `pulsar initialize-cluster-metadata`. This is assuming a pristine cluster. If there's already BK initialized, it might not initialize the required Pulsar metadata.
